nntrainer/src/neuralnet.cpp

```diff
@@ -467,21 +271,52 @@ int NeuralNetwork::loadFromConfig() {
       break;
     }
 
-    /** Add activation layer */
-    const char *acti_str =
-      iniparser_getstring(ini, (layer_name + ":Activation").c_str(), unknown);
-    ActiType act = (ActiType)parseType(acti_str, TOKEN_ACTI);
-    layers.back()->setActivation(act);
+    unsigned int property_end =
+      static_cast<unsigned int>(Layer::PropertyType::unknown);
+
+    for (unsigned int i = 0; i < property_end; ++i) {
+      std::string prop = propToStr(i);
+      std::string value =
+        iniparser_getstring(ini, (layer_name + ":" + prop).c_str(), "");
+
+      try {
+        /// if problem setting property, it will throw std::invalid_argument
+        layer->setProperty(static_cast<Layer::PropertyType>(i), value);
+      } catch (std::out_of_range &e) {
+        /// intended
+      }
+    }
+
+    status = layer->setName(layer_name);
+    NN_INI_RETURN_STATUS();
 
-    /** Add flatten layer */
-    bool flatten =
-      iniparser_getboolean(ini, (layer_name + ":Flatten").c_str(), false);
-    layers.back()->setFlatten(flatten);
+    status = addLayer(layer);
+    NN_INI_RETURN_STATUS();
+  }
+
+  /**< Additional validation and handling for the neural network */
+  if (iniparser_find_entry(ini, "network") == 0) {
```

Comment:
how about doing in this the start, and calling `loadNetworkConfig((void *)ini);` in the else of this than in the loop above.
may be same of the data buffer - move outside the above loop.

Reply:
moved this before loop